topic_dynamics/parsing.py

```diff
@@ -0,0 +1,420 @@
+"""
+Parsing-related functionality.
+"""
+
+from collections import Counter, namedtuple
+import datetime
+from operator import itemgetter
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Any, List, Tuple
+
+from tqdm import tqdm
+import tree_sitter
+
+from .parsers.utils import get_parser
+from .slicing import get_dates, checkout_by_date
+
+NODE_TYPES = {"c": ["identifier", "type_identifier"],
+              "c-sharp": ["identifier", "type_identifier"],
+              "cpp": ["identifier", "type_identifier"],
+              "java": ["identifier", "type_identifier"],
+              "python": ["identifier", "type_identifier"]}
+
+SliceLine = namedtuple("SliceLine", "date start_index end_index")
+TokenLine = namedtuple("TokenLine", "index address tokens")
+
+
+def slices_to_int(slices_line: List[str]) -> List[Any]:
+    """
+    Transform numerals in the split line of Slices file into an 'int' type during initialization.
+    :param slices_line: a split List[str, str, str].
+    :return: transformed List[str, int, int].
+    """
+    return [slices_line[0], int(slices_line[1]), int(slices_line[2])]
+
+
+def tokens_to_int(tokens_line: List[str]) -> List[Any]:
+    """
+    Transform numerals in the split line of Tokens file into an 'int' type during initialization.
+    :param tokens_line: a split List[str, str, str].
+    :return: transformed List[int, str, str].
+    """
+    return [int(tokens_line[0]), tokens_line[1], tokens_line[2]]
+
+
+def get_extensions(lang: str) -> str:
+    """
+    Returns the extension for a given language. TODO: get rid of this and add enry.
+    :param lang: language name.
+    :return: the extension.
+    """
+    extensions = {"cpp": "cpp",
+                  "java": "java",
+                  "python": "py"}
+    return extensions[lang]
+
+
+def get_files(directory: str, extension: str) -> List[str]:
+    """
+    Get a list of files with a given extension.
+    :param directory: the root directory that is studied.
```

Comment:
```suggestion
    :param directory: path to directory.
```
This function should know nothing about it's context to use - even in description.